app/models/post.rb

```diff
@@ -13,6 +13,6 @@ def self.update_posts_counter(user)
   end
 
   def five_most_recent_comments
-    Comment.where(post_id: id).order(updated_at: :desc).limit(5).sort_by(&:updated_at)
+    comments.includes(:author).order(updated_at: :desc).limit(5).sort_by(&:updated_at)
   end
 end
```

Comment:
### Although I am highlighting this file, the proposed change should be made in the `Posts` controller for the `index` action:
- [ ] To eliminate the N+1 query problem, I would advise you to create a variable for the post like this:
```ruby
@posts = @user.posts.includes(:comments)
```

Then you can use the @posts variable to replace the `@user.posts.sort` statement in the `app/views/posts/index.html.erb` file. You can try to create many posts and comments in the `db/seeds.rb` file and when you navigate to the `Posts#Index` page, you would see the difference in the database query made in the terminal. 👍 💻 

Kindly implement this change. 😎

Reply:
Hi @Gambit142, thanks for your review! I'll add the change, but if I were to install `bullet` and navigate to the post's, index page, then it throws me this alert:
![image](https://github.com/dicodiaz/rails-blog/assets/70329406/b38d34f5-26c2-4ab1-aa58-4bd289739d97)
Taking this into account, I think this is an example of eager loading that we should avoid.

Reply:
Hi @dicodiaz, please do not add the `bullet` gem. It was stated in your requirements that you should not use the `bullet` gem. The essence of the exercise is to practise removing **_N + 1 queries_** by writing better codes. Please do not use the `bullet` gem. 👍

Reply:
Gotcha, I was just stating what would happen if I did. I will still add the suggested code, thank you very much.